[Jason Ng] iP

jason-ng-zq99 opened the pull request:
# Duke
> "Your mind is for having ideas, not holding them." - David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

Duke frees your mind of having to remember things you need to do. It's, 

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use 

All you need to o is,
1. download it from [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add yourt asks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:

- [X] Managing tasks
- [X] Managing deadlines 
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the ```main``` method: 

```javascript
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

jason-ng-zq99 committed:
Add ability to echo commands, and terminates on "bye"

jason-ng-zq99 committed:
Add new class Printer

jason-ng-zq99 committed:
Add Printer class

jason-ng-zq99 committed:
Change print message

jason-ng-zq99 committed:
Add functionality to take and store input in list

jason-ng-zq99 committed:
Add PrintList function to Printer class
Add ability to list for Level-2

jason-ng-zq99 committed:
Add new class Task
New class helps to print tasks better as well

jason-ng-zq99 committed:
Add feature to mark tasks as done
Also prints message regarding the marked task

jason-ng-zq99 committed:
Add 3 new classes
Todo, Deadline, Event
All inherit from superclass Task

jason-ng-zq99 committed:
Update list capabilities
List now able to store Todos, Deadlines and Events

jason-ng-zq99 committed:
Update todo, deadline, event features
List now able to properly print messages for above 3 features

jason-ng-zq99 committed:
Edit text-ui-test folder
Updated ACTUAL.TXT and EXPECTED.TXT to match current update to Duke app

jason-ng-zq99 committed:
Add new class DukeException
Also updated to check for todo without a task trailing the command

jason-ng-zq99 committed:
Update errors thrown with incorrect inputs

jason-ng-zq99 committed:
Edit printed message for list command

jason-ng-zq99 committed:
Add exception handling for Deadline and Event

jason-ng-zq99 committed:
Fix bug at event
Also edited EXPECTED.txt for new outputs

jason-ng-zq99 committed:
Update error handling for event
Able to handle for a space character for event date

jason-ng-zq99 committed:
Update input.txt and Expected.txt
To account for new error messages

jason-ng-zq99 committed:
Add new feature delete

Able to now delete tasks off the tasklist

jason-ng-zq99 committed:
Update delete feature

Delete feature now prints what is being deleted,
and number of tasks left

jason-ng-zq99 committed:
Update delete

Delete feature now handles errorneous number inputs that are out of range

jason-ng-zq99 committed:
Update test files

New outputs from new delete feature

jason-ng-zq99 committed:
Edit with ENUM

Use enum for special words:
* bye
* done
* list
* todo
* deadline
* event
* delete

jason-ng-zq99 committed:
Add logo attribute for Todo, Deadline, Event

jason-ng-zq99 committed:
Add Storage class and Save method

Storage class will contain methods for saving and loading tasks
Save method saves tasks into hard disk for usage

jason-ng-zq99 committed:
Updated storage txt file

jason-ng-zq99 committed:
Update Storage class

Add Load method to load previous tasks.

Handles the event where no past tasks available.

jason-ng-zq99 committed:
Update classes Event, Deadline, Todo

Edit markDone() method to include logo for printed message

jason-ng-zq99 committed:
Update special tasks deadline, event

Special tasks now constructed with a datetime instead of a string

jason-ng-zq99 committed:
Update Deadline and Event feature

Deadline and Event now accepts and stores date

Able to print stored date as well

jason-ng-zq99 committed:
Update Storage.java

Edit append newline with '\n' to System.lineSeparator

jason-ng-zq99 committed:
Update Deadline and Event feature

Now able to accept timing also on top of date

jason-ng-zq99 committed:
Merge pull request #1 from jason-ng-zq99/branch-Level-7

Merge branch-Level-7

jason-ng-zq99 committed:
Merge pull request #2 from jason-ng-zq99/branch-Level-8

Merge branch-Level-8

jason-ng-zq99 committed:
Refactor Printer class to Ui

jason-ng-zq99 committed:
Add new class TaskList

TaskList class is in charge of all tasklist operations

jason-ng-zq99 committed:
Add new class Parser

Parser class deals with making sense of user command
Code is generally refactored to make better usage of all the classes

jason-ng-zq99 committed:
Refactor code

Organize into following packages under duke:
* commands
* data
* parser
* storage
* ui
* main

jason-ng-zq99 committed:
Add test folders

jason-ng-zq99 committed:
Add test module

Include some JUnit test cases

jason-ng-zq99 committed:
Update .gitignore

Include jar file in gitignore

jason-ng-zq99 committed:
Add Javadocs

For all methods and classes

jason-ng-zq99 committed:
Edit code to follow the coding standard.

jason-ng-zq99 committed:
Add Find class

Find class handles the find command to find tasks related to input word

jason-ng-zq99 committed:
Add new methods to Parser class to support Find command

New methods are:
* isFind - helps to check if input command is a "find" command
* parseFind - parses a "find" command

jason-ng-zq99 committed:
Add new method to Ui class to support "find" command

New method is:
* PrintFind - prints the list of matched tasks found under "find" command

jason-ng-zq99 committed:
Update Main.java

run() method in Main.java now supports "find" commands

jason-ng-zq99 committed:
Merge branch 'branch-A-CodingStandard'

Update for A-CodingStandard

jason-ng-zq99 committed:
Merge branch 'branch-Level-9'

For Duke Level-9

jason-ng-zq99 committed:
Update Javadocs

For new methods and main.java

jason-ng-zq99 commented:
# Duke
> "Your mind is for having ideas, not holding them." - David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

Duke frees your mind of having to remember things you need to do. It's, 

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use 

All you need to o is,
1. download it from [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add yourt asks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:

- [X] Managing tasks
- [X] Managing deadlines 
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the ```main``` method: 

```javascript
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

jason-ng-zq99 committed:
Merge pull request #3 from jason-ng-zq99/add-gradle-support

Add Gradle support

jason-ng-zq99 committed:
Update build.gradle

Change mainClassName to fit current project structure

jason-ng-zq99 committed:
Merge branch 'branch-A-Gradle'

For A-Gradle

jason-ng-zq99 committed:
Create new directories and add checkstyle.xml

Allow Gradle to run checkstyle

jason-ng-zq99 committed:
Add suppressions.xml

To support checkstyle

jason-ng-zq99 committed:
Edit Main.java

Based on violations flagged out by checkstyle

jason-ng-zq99 committed:
Merge branch 'branch-A-Checkstyle'

Support for Checkstyle

jason-ng-zq99 committed:
Add dependencies to build.gradle

Support for JavaFX

jason-ng-zq99 committed:
Edit Main.java

Main class now extends Application and overrides start method to run GUI

jason-ng-zq99 committed:
Add new Launcher class

Also add start method in Main.java to support javaFX

jason-ng-zq99 committed:
Create anchorpane for Duke GUI

jason-ng-zq99 committed:
Increase width of send button

jason-ng-zq99 committed:
Add new class DialogBox

jason-ng-zq99 committed:
Add images to dialogue box

jason-ng-zq99 committed:
Edit chat box

User and system input now at different sides

ChuangZheQuan commented on src/main/java/duke/data/TaskList.java:
```diff
@@ -0,0 +1,89 @@
+     *
+     * @return The entire taskList.
+     */
+    public ArrayList<Task> getEntire() {
```
Maybe can be rename to ```getAllTasks``` just to be a little more descriptive?

chownces replied:
Agree!

ChuangZheQuan commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,106 @@
+     *
+     * @param message The message to be printed.
+     */
+    public void PrintMessage(String message) {
```
I think method names should start with a lower case. Same for all the other methods in this class.

chownces replied:
Agreed!

https://se-education.org/guides/conventions/java/intermediate.html#naming

ChuangZheQuan commented on src/main/java/duke/commands/Event.java:
```diff
@@ -0,0 +1,70 @@
+ */
+public class Event extends Task {
+    /** Event time of the task */
+    protected LocalDateTime at;
```
Maybe ```at``` should be more descriptive?

ChuangZheQuan commented on src/main/java/duke/commands/Deadline.java:
```diff
@@ -0,0 +1,70 @@
+ */
+public class Deadline extends Task {
+    /** Deadline of the task */
+    protected LocalDateTime by;
```
Maybe ```by``` should be a little more descriptive?

ChuangZheQuan reviewed (commented):
Great job! Not many naming errors, just the method names in your Ui class should start with lower cases!

jason-ng-zq99 committed:
Add new directory main/resources/view

Add DialogBox.fxml and MainWindow.fxml

jason-ng-zq99 committed:
Refactor main into main package

jason-ng-zq99 committed:
Rebuilt using FXML

chownces commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,280 @@
+     * @param input The input string.
+     * @return True if it is a "list" command, false otherwise.
+     */
+    public boolean isList(String input) {
+        return input.equals(SPECIAL_TASK.list.name());
+    }
```
This name might be a little misleading. Perhaps `isListCommand`?

chownces commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,118 @@
+                String description = splitLine.get(2);
+
+                Task toAdd;
+                switch (taskType) {
+                    case "T":
+                        toAdd = new Todo(description);
+                        break;
+                    case "D":
+                        LocalDateTime by = LocalDateTime.parse(splitLine.get(3));
+                        toAdd = new Deadline(description, by);
+                        break;
+                    case "E":
+                        LocalDateTime at = LocalDateTime.parse(splitLine.get(3));
+                        toAdd = new Event(description, at);
+                        break;
+                    default:
+                        toAdd = new Task(description);
+                        break;
+                }
```
Do follow the module coding style for switch block indentations!

https://se-education.org/guides/conventions/java/intermediate.html#layout

chownces commented on src/test/java/duke/data/TaskListTest.java:
```diff
@@ -0,0 +1,22 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TaskListTest {
+    @Test
+    public void testDelete() {
+        ArrayList<Task> test = new ArrayList<>();
+        Task task = new Task("test");
+        test.add(task);
+        test.add(task);
+        test.add(task);
+        TaskList testList = new TaskList(test);
+        testList.delete(1);
+        assertEquals(2, testList.size());
+    }
```
I feel that the test method name can be more descriptive. Perhaps `taskDeletion_nonEmptyTaskList_successful`? Similarly for the other tests as well

chownces reviewed (commented):
Overall great work! Do take note of the coding standard violations mentioned! Feel free to take up the suggestions too :)

jason-ng-zq99 committed:
Change ui package

Print methods now returns a string instead

This helps support GUI

jason-ng-zq99 committed:
Remove unnecessary code in Main.java

jason-ng-zq99 committed:
Merge branch 'branch-Level-10'

Implement GUI support using javaFX

Now able to take in commands previously only available in CLI version of Duke

jason-ng-zq99 committed:
Update Find class

Method findMatch now takes in varargs as parameter to search for tasks that better matches input words

jason-ng-zq99 committed:
Merge branch 'branch-A-Varargs'

Supports A-Varargs

jason-ng-zq99 committed:
Update Javadocs

Update Javadocs for all files with recent changes

jason-ng-zq99 committed:
Remove borders in chat messages

Duke GUI has message borders from the previous CLI edition.

These borders are too long and looks unnecessary in the GUI edition.

Removing the borders will make the GUI chat box cleaner and easier
to read lengthy tasks.

We can make shorter borders that fit but we can always do that through javaFX.

jason-ng-zq99 committed:
Bye command: make program terminate

Bye command is not terminating the program.

Program still runs and window remains open after executing bye command,
which is not what we expect it to do.

Let's add a Platform.close() call to getResponse. It will only be
called when user gives a Bye command.

This way, the user need not terminate the program by him or herself.

jason-ng-zq99 committed:
Edit code to fit checkstyle

jason-ng-zq99 committed:
Add assertions

Assertions to check at places where there is input, font usage
and loading of files

jason-ng-zq99 committed:
Refactor code

Improve code quality

jason-ng-zq99 committed:
Use streams in load() function in Storage.java

jason-ng-zq99 committed:
Merge pull request #4 from jason-ng-zq99/branch-A-Assertions

Add assertions

jason-ng-zq99 committed:
Merge branch 'master' into branch-A-CodeQuality

jason-ng-zq99 committed:
Merge branch 'master' into branch-A-Streams

jason-ng-zq99 committed:
Merge pull request #5 from jason-ng-zq99/branch-A-CodeQuality

Branch a code quality

jason-ng-zq99 committed:
Merge branch 'master' into branch-A-Streams

jason-ng-zq99 committed:
Merge pull request #6 from jason-ng-zq99/branch-A-Streams

Branch a streams

jason-ng-zq99 committed:
Add files for GitHub actions to setup CI

Add new folder workflows to contain .yml for CI to run

jason-ng-zq99 committed:
Update search function

jason-ng-zq99 committed:
Merge branch 'branch-C-BetterSearch'

jason-ng-zq99 committed:
Update mainClassName for Gradle

jason-ng-zq99 committed:
Edit gradle file to enable assertions

jason-ng-zq99 committed:
Update new photos for duke and user

jason-ng-zq99 committed:
Edit termination of app

jason-ng-zq99 committed:
Merge branch 'branch-A-BetterGui'

jason-ng-zq99 committed:
Add Ui.png and brief user guide for app

jason-ng-zq99 committed:
Add space for Feature under README.md

jason-ng-zq99 committed:
Edit Ui.png and dialogbox height

jason-ng-zq99 committed:
Fix delete bug where first task on list could not be deleted

jason-ng-zq99 committed:
Edit typo in README.md

jason-ng-zq99 committed:
Merge branch 'branch-editReadme'
